crates/autopilot/src/solvable_orders.rs

```diff
@@ -234,6 +228,8 @@ impl SolvableOrdersCache {
         let removed = counter.record(&auction.orders);
         filtered_order_events.extend(removed);
 
+        let auction =
```

Comment:
Why even create a `model::Auction` two lines above? I think we should just take the orders we have convert those into the domain model and then create the auction using the domain code (applying fees is domain logic).

Ideally we would get rid of `self.database.solvable_orders` above so this entire method can operate on domain models, but I assume this would make the refactor even larger. Can we capture this as a follow up task?